### ISSUE TITLE
STATUS_CARD displays only 1 or 0

### ISSUE THREAD

**Issue by Daemon256** [ayushsharma82/ESP-DASH#107]
The screenshot shows that the STATUS_CARD displays the inscription "working". I can't do it. I only receive either 0 or 1

```
PLATFORM: Espressif 8266 (2.6.3+sha.2e5a2c4) > NodeMCU 1.0 (ESP-12E Module)
HARDWARE: ESP8266 80MHz, 80KB RAM, 4MB Flash
PACKAGES:
 - framework-arduinoespressif8266 3.20704.0 (2.7.4)
 - tool-esptool 1.413.0 (4.13)
 - tool-esptoolpy 1.30000.201119 (3.0.0)
 - tool-mklittlefs 1.203.210203 (2.3)
 - tool-mkspiffs 1.200.0 (2.0)
 - toolchain-xtensa 2.40802.200502 (4.8.2)
Converting sun2000gtil.ino
LDF: Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 52 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <Streaming> 6.0.8+sha.faef8e7
|-- <TaskScheduler> 3.2.2+sha.31c3889
|-- <ArduinoJson> 6.17.3+sha.22904a48
|-- <Average> 1.0.17+sha.81630b6
|-- <Time> 1.6.0+sha.410aee4
|-- <UniversalTelegramBot> 1.3.0+sha.65e6f82
|   |-- <ArduinoJson> 6.17.3+sha.22904a48
|-- <ESP-DASH> 3.0.3+sha.edd044b
|   |-- <ESP Async WebServer> 1.2.3
|   |   |-- <ESPAsyncTCP> 1.2.2
|   |   |   |-- <ESP8266WiFi> 1.0
|   |   |-- <Hash> 1.0
|   |   |-- <Time> 1.6.0+sha.410aee4
|   |   |-- <ESP8266WiFi> 1.0
|   |   |-- <ArduinoJson> 6.17.3+sha.22904a48
|   |-- <ArduinoJson> 6.17.3+sha.22904a48
|   |-- <ESP8266WiFi> 1.0
|   |-- <ESPAsyncTCP> 1.2.2
|   |   |-- <ESP8266WiFi> 1.0
|-- <ESPAsyncTCP> 1.2.2+sha.1547686
|   |-- <ESP8266WiFi> 1.0
|-- <ESP Async WebServer> 1.2.3+sha.f6fff3f
|   |-- <ESPAsyncTCP> 1.2.2
|   |   |-- <ESP8266WiFi> 1.0
|   |-- <Hash> 1.0
|   |-- <ArduinoJson> 6.17.3+sha.22904a48
|   |-- <ESP8266WiFi> 1.0
|   |-- <Time> 1.6.0+sha.410aee4
|-- <ESP8266WiFi> 1.0
|-- <ESPAsyncTCP> 1.2.2
|   |-- <ESP8266WiFi> 1.0
|-- <ESP Async WebServer> 1.2.3
|   |-- <ESPAsyncTCP> 1.2.2
|   |   |-- <ESP8266WiFi> 1.0
|   |-- <Hash> 1.0
|   |-- <Time> 1.6.0+sha.410aee4
|   |-- <ESP8266WiFi> 1.0
|   |-- <ArduinoJson> 6.17.3+sha.22904a48
```

```
//  Dashboard Cards 
//  Format - (Dashboard Instance, Card Type, Card Name, Card Symbol(optional) )

Card dutycycle_card(&dashboard, PROGRESS_CARD, "Dutycycle grid", "", 0, 100);
Card power_card(&dashboard, GENERIC_CARD, "Power house");
Card battery_voltage_card(&dashboard, GENERIC_CARD, "Battery voltage");
Card charger_status_card(&dashboard, STATUS_CARD, "status", "danger");
```
```

/* Update Card Values */
  
  dutycycle_card.update((int)dutycycle);
  power_card.update((int)dataJsonPower.json_parse_value_power);
  battery_voltage_card.update((float)dataJsonPower.json_parse_value_voltage);
  charger_status_card.update("status", "danger"); //(i ? "Charger status ON" : "Charger status OFF", i ? "success" : "danger");
  //battery_bms_card.update(error_bat ? "Charger status ON" : "Charger status OFF", error_bat ? "success" : "danger");

  /* Send Updates to our Dashboard (realtime) */
  dashboard.sendUpdates();
```

**Comment by Daemon256** [comment]
![11](https://user-images.githubusercontent.com/7066429/109057042-4c590e00-76f2-11eb-8215-883f70a59bf8.png)

**Comment by Daemon256** [comment]
`Card charger_status_card(&dashboard, STATUS_CARD, "status", "success");`
`charger_status_card.update("test", "success");`
`dashboard.sendUpdates();`
result:
![12](https://user-images.githubusercontent.com/7066429/109112723-25c7c100-774c-11eb-887f-d26603cc8a61.png)

`charger_status_card.update(0, "success");`
result:
![13](https://user-images.githubusercontent.com/7066429/109112980-8a831b80-774c-11eb-9531-fc3f64c64bed.png)

**Comment by ayushsharma82** [comment]
Weird. I'll test this out in a day or two.

**Comment by Daemon256** [comment]
only this helped
`  charger_status_card.update(i ? (String) "charged" : (String) "disconnected", i ? (const char *)success : (const char *)idle);`

But I could not solve the problem with a full strip
`
Card dutycycle_card(&dashboard, PROGRESS_CARD, (const char *)"Dutycycle grid", (const char *)"", (int)0, (int)135);
dutycycle_card.update((int)dutycycle, (const char *)"");`

Any value draws a full strip

**Comment by ayushsharma82** [comment]
@Daemon256  I've tried to fix the status card problem, can you try your code with the library in this tree: https://github.com/ayushsharma82/ESP-DASH/tree/string-fix

**Comment by Daemon256** [comment]
`charger_status_card.update(i ? "charged" : "disconnected", i ? success : idle);`
Working

But I could not solve the problem with a full strip
`Card dutycycle_card(&dashboard, PROGRESS_CARD, (const char *)"Dutycycle grid", (const char *)"", (int)0, (int)135); dutycycle_card.update((int)dutycycle, (const char *)"");`

Any value "dutycycle" draws a full strip

And icons do not change until the page is refreshed

**Comment by ayushsharma82** [comment]
Great. Thanks for your effort in testing it.

I'll take a look over progress card tomorrow and fix it.

**Comment by ayushsharma82** [comment]
Fixed status card icon reactivity with 276089d

Can you tell me the data type of `dutycycle`? I tested the progress card roughly and found no issues with it.

**Comment by Daemon256** [comment]
> Fixed status card icon reactivity with [276089d](https://github.com/ayushsharma82/ESP-DASH/commit/276089d074beb29e4b530877d63bb5f0a7eb5c14)
> 
> Can you tell me the data type of `dutycycle`? I tested the progress card roughly and found no issues with it.

`Card dutycycle_card(&dashboard, PROGRESS_CARD, (const char *)"Dutycycle grid", (const char *)"", (int)0, (int)135);
dutycycle_card.update(25, (const char *)"");`

![14](https://user-images.githubusercontent.com/7066429/111744016-9cfbfb00-889b-11eb-9d21-1ebae4ffe6a8.png)

**Comment by ayushsharma82** [comment]
I'll check it practically with a ESP32.

**Comment by Daemon256** [comment]
> Fixed status card icon reactivity with [276089d](https://github.com/ayushsharma82/ESP-DASH/commit/276089d074beb29e4b530877d63bb5f0a7eb5c14)

Working !

**Comment by Daemon256** [comment]
> I'll check it practically with an ESP.

Were you able to check?

**Comment by ayushsharma82** [comment]
@Daemon256  Fixed with 849523e. Sorry, I was quite busy these days.

**Comment by ayushsharma82** [comment]
Feel free to close the issue if it was resolved.

Thanks
